drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -967,6 +973,144 @@ static int siwx91x_dev_init(const struct device *dev)
 	return 0;
 }
 
+static int siwx91x_convert_z_sl_twt_req_type(enum wifi_twt_setup_cmd z_req_cmd)
+{
+	switch (z_req_cmd) {
+	case WIFI_TWT_SETUP_CMD_REQUEST:
+		return REQUEST_TWT;
+	case WIFI_TWT_SETUP_CMD_SUGGEST:
+		return SUGGEST_TWT;
+	case WIFI_TWT_SETUP_CMD_DEMAND:
+		return DEMAND_TWT;
+	default:
+		return -EINVAL;
+	}
+}
+
+static int siwx91x_set_twt_setup(struct wifi_twt_params *params, sl_wifi_twt_request_t *twt_req)
+{
+	sl_status_t status = SL_STATUS_OK;
+
+	if (!params->setup.twt_info_disable) {
+		params->fail_reason = WIFI_TWT_FAIL_OPERATION_NOT_SUPPORTED;
+		return -ENOTSUP;
+	}
+
+	if (params->setup.responder) {
+		params->fail_reason = WIFI_TWT_FAIL_OPERATION_NOT_SUPPORTED;
+		return -ENOTSUP;
+	}
+
+	/* implicit -> won't do renegotiation
+	 * explicit -> must do renegotiation for each session
+	 */
+	if (params->setup.implicit) {
+		twt_req->implicit_twt = 1;
+	} else {
+		/* explicit twt is not supported */
+		params->fail_reason = WIFI_TWT_FAIL_OPERATION_NOT_SUPPORTED;
+		return -ENOTSUP;
+	}
+
+	twt_req->twt_enable = 1;
+	twt_req->wake_int_exp = params->setup.twt_exponent;
+	twt_req->triggered_twt = params->setup.trigger;
+	twt_req->un_announced_twt = !params->setup.announce;
+	twt_req->wake_int_mantissa = params->setup.twt_mantissa;
+	twt_req->wake_duration_unit = 0;
+	twt_req->wake_duration = params->setup.twt_wake_interval;
+
+	if (params->setup.twt_wake_interval > 255 * 256) {
+		twt_req->wake_duration_unit = 1;
+		twt_req->wake_duration = params->setup.twt_wake_interval / 256;
+	} else {
+		twt_req->wake_duration_unit = 0;
+		twt_req->wake_duration = params->setup.twt_wake_interval / 1024;
+	}
+
+	status = sl_wifi_enable_target_wake_time(twt_req);
+	if (status != SL_STATUS_OK) {
+		params->fail_reason = WIFI_TWT_FAIL_CMD_EXEC_FAIL;
+		params->resp_status = WIFI_TWT_RESP_NOT_RECEIVED;
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int siwx91x_set_twt_teardown(struct wifi_twt_params *params, sl_wifi_twt_request_t *twt_req)
+{
+	sl_status_t status = SL_STATUS_OK;
+
+	twt_req->twt_enable = 0;
```

Comment:
You have already assigned 0 to `twt_enable` (but since this has been made in the caller function, this is not easy to follow).